pkg/scalers/splunk_o11y_scaler.go

```diff
@@ -0,0 +1,264 @@
+		default:
+			return nil, fmt.Errorf("queryAggregator value %s has to be one of 'max', 'min', or 'avg'.", queryAggregator)
+		}
+	} else {
+		meta.queryAggegrator = ""
+	}
+
+	// accessToken
+	if accessToken, ok := config.AuthParams["accessToken"]; ok {
+		// cleanedAccessToken := strings.ReplaceAll(accessToken, "\n", "")
+		// cleanedAccessToken = strings.ReplaceAll(accessToken, "\r", "")
+		meta.accessToken = accessToken
+	} else {
+		return nil, fmt.Errorf("no accessToken given")
+	}
+
+	// realm
+	if realm, ok := config.TriggerMetadata["realm"]; ok {
+		meta.realm = realm
+	} else {
+		return nil, fmt.Errorf("no realm given")
+	}
+	logger.Info("Splunk Realm -> %s\n", meta.realm)
+
+	// Debug TODO check
+	meta.vType = v2.ValueMetricType
+
+	return &meta, nil
+}
+
+func newSplunkO11yConnection(ctx context.Context, meta *splunkO11yMetadata, config *scalersconfig.ScalerConfig) (*signalflow.Client, error) {
+	accessToken := meta.accessToken
+	realm := meta.realm
+
+	if realm == "" || accessToken == "" {
+		return nil, fmt.Errorf("error. could not find splunk access token or ream.")
+	}
+
+	apiClient, err := signalflow.NewClient(
+		signalflow.StreamURLForRealm(realm),
+		signalflow.AccessToken(accessToken))
+	if err != nil {
+		return nil, fmt.Errorf("error creating SignalFlow client: %w", err)
+	}
+
+	return apiClient, nil
+}
+
+func logMessage(logger logr.Logger, msg string, value float64) {
+	if value != -1 {
+		msg = fmt.Sprintf("splunk_o11y_scaler: %s -> %v", msg, value)
+	} else {
+		msg = fmt.Sprintf("splunk_o11y_scaler: %s", msg)
+	}
+	logger.Info(msg)
+}
+
+func (s *splunkO11yScaler) getQueryResult(ctx context.Context) (float64, error) {
+	// var duration time.Duration = 1000000000 // one second in nano seconds
+	var duration time.Duration = 10000000000 // ten seconds in nano seconds
+
+	comp, err := s.apiClient.Execute(context.Background(), &signalflow.ExecuteRequest{
+		Program: s.metadata.query,
+	})
+	if err != nil {
+		return -1, fmt.Errorf("error: could not execute signalflow query: %w", err)
+	}
+
+	go func() {
+		time.Sleep(duration)
+		if err := comp.Stop(context.Background()); err != nil {
+			s.logger.Info("Failed to stop computation")
+		}
+	}()
+
+	logMessage(s.logger, "Received Splunk Observability metrics", -1)
+
+	max := math.Inf(-1)
+	min := math.Inf(1)
+	valueSum := 0.0
+	valueCount := 0
+	s.logger.Info("getQueryResult -> Now Iterating")
+	for msg := range comp.Data() {
+		s.logger.Info("getQueryResult -> msg: %+v\n", msg)
+		if len(msg.Payloads) == 0 {
+			s.logger.Info("getQueryResult -> No data retreived. Continuing")
+			continue
+		}
+		for _, pl := range msg.Payloads {
+			value, ok := pl.Value().(float64)
+			if !ok {
+				return -1, fmt.Errorf("error: could not convert Splunk Observability metric value to float64")
+			}
+			logMessage(s.logger, "Encountering value ", value)
+			if value > max {
+				max = value
+			}
+			if value < min {
+				min = value
+			}
+			valueSum += value
+			valueCount++
+		}
+	}
```

Comment:
If you decide to use Splunk-side aggregation, then this loop would be significantly simpler and you should expect one and only one value.